tests/src/androidTest/java/com/squareup/duktape/DuktapeSetTest.java

```diff
@@ -494,4 +497,191 @@ public String getValue() {
+        return args;
+      }
+    });
+
+    assertThat((Object[]) duktape.evaluate("Sorter.sort(['b', 'd', 'c', 'a'])"))
+        .isEqualTo(new String[]{ "a", "b", "c", "d" });
+
+    try {
+      duktape.evaluate("Sorter.sort(['b', 'd', 3, 'a'])");
+      fail();
+    } catch (DuktapeException expected) {
+      assertThat(expected).hasMessage("TypeError: string required, found 3 (stack index -1)");
+    }
+  }
+
+  interface DoubleSorter {
+    double[] sort(double[] args);
+    Double[] sortNullsFirst(Double[] args);
+  }
+
+  @Test public void arraysOfDoubles() {
+    duktape.set("Sorter", DoubleSorter.class, new DoubleSorter() {
+      @Override public double[] sort(double[] args) {
+        Arrays.sort(args);
+        return args;
+      }
+      @Override public Double[] sortNullsFirst(Double[] args) {
+        Arrays.sort(args, new Comparator<Double>() {
+          @Override public int compare(Double l, Double r) {
+            if (l == null) return -1;
+            if (r == null) return 1;
+            return l.compareTo(r);
+          }
+        });
+        return args;
+      }
+    });
+
+    assertThat((Object[]) duktape.evaluate("Sorter.sort([2.9, 2.3, 3, 1])"))
+        .isEqualTo(new Object[]{ 1.0, 2.3, 2.9, 3.0 });
+
+    try {
+      duktape.evaluate("Sorter.sort([2.3, 4, null, 1])");
+      fail();
+    } catch (DuktapeException expected) {
+      assertThat(expected).hasMessage("TypeError: number required, found null (stack index -1)");
+    }
+
+    assertThat((Object[]) duktape.evaluate("Sorter.sortNullsFirst([2.9, null, 3, 1])"))
+        .isEqualTo(new Object[]{ null, 1.0, 2.9, 3.0 });
+  }
+
+  interface IntSorter {
+    int[] sort(int[] args);
+    Integer[] sortNullsFirst(Integer[] args);
+  }
+
+  @Test public void arraysOfInts() {
+    duktape.set("Sorter", IntSorter.class, new IntSorter() {
+      @Override public int[] sort(int[] args) {
+        Arrays.sort(args);
+        return args;
+      }
+
+      @Override public Integer[] sortNullsFirst(Integer[] args) {
+        Arrays.sort(args, new Comparator<Integer>() {
+          @Override public int compare(Integer l, Integer r) {
+            if (l == null) return -1;
+            if (r == null) return 1;
+            return l.compareTo(r);
+          }
+        });
+        return args;
+      }
+    });
+
+    assertThat((Object[]) duktape.evaluate("Sorter.sort([2, 4, 3, 1])"))
+      .isEqualTo(new Object[]{ 1.0, 2.0, 3.0, 4.0 });
+
+    try {
+      duktape.evaluate("Sorter.sort([2, 4, null, 1])");
+      fail();
+    } catch (DuktapeException expected) {
+      assertThat(expected).hasMessage("TypeError: number required, found null (stack index -1)");
+    }
+
+    assertThat((Object[]) duktape.evaluate("Sorter.sort([2, 4, 3.14, 1])"))
+        .isEqualTo(new Object[]{ 1.0, 2.0, 3.0, 4.0 });
+
+    assertThat((Object[]) duktape.evaluate("Sorter.sortNullsFirst([2, null, 3, 1])"))
+        .isEqualTo(new Object[]{ null, 1.0, 2.0, 3.0 });
```

Comment:
curious that your comparator puts null first, whereas on DuktapeGetTest:501 puts it last